Added City Weather view

kryzhikh opened the pull request:
<img src="https://user-images.githubusercontent.com/1997070/179602264-5eda9f38-5468-4a66-834b-4444d74fe4f7.png" width="50%" height="50%">

kotofey committed:
Added City Weather view

MariaKulkova commented on WeatherUI/Params/Temperature.swift:
```diff
@@ -0,0 +1,33 @@
+
+import Foundation
+
+struct Temperature: Hashable {
```
Realised that there are native Units and Measurements on iOS that could be really handy there

kryzhikh replied:
that's awesome, switched to Measurements!

MariaKulkova commented on WeatherUI/Params/WeatherConditions.swift:
```diff
@@ -0,0 +1,97 @@
+    var imageName: String {
+        switch self {
+        case .clearSky:
+            return "sun_35"
```
If the images are named in similar style, maybe we can use suffixes to get thumbs and big images?

kryzhikh replied:
updated

MariaKulkova commented on WeatherUI/Views/CityWeatherView/CityWeatherView.swift:
```diff
@@ -0,0 +1,98 @@
+                    Text(model.cityName)
+                        .font(.system(size: 18, weight: .regular))
+                    Image("city_pin")
+                }
+                HStack(alignment: .center) {
```
Is this a common style? I'd prefer to separate views with one empty line, but it's the matter of taste I guess

kryzhikh replied:
thats a good point about swiftui style guide. i didn't find any, although we can have some essential agreements towards this.
speaking of this particular point - it is arguable. basically most of entities of swiftui are views, but i believe we shouldn't separate each with empty line. we could separate some grouping views, like lists, grids, etc. but i'm not sure this gives a lot more readability since grouping views are already noticeable due to indentation.

MariaKulkova commented on WeatherUI/Views/CityWeatherView/CityWeatherView.swift:
```diff
@@ -0,0 +1,98 @@
+                    Image("city_pin")
+                }
+                HStack(alignment: .center) {
+                    VStack(alignment: .leading, spacing: 8.0) {
```
Why 8.0 and not just 8?

kryzhikh replied:
that automatically gets this way when adjusting attributes using attributes inspector. i suggest leaving this kind of things as is unless we have some style agreements

MariaKulkova commented on WeatherUI/Views/CityWeatherView/CityWeatherView.swift:
```diff
@@ -0,0 +1,98 @@
+                    Spacer()
+                    Label("\(model.humidity) %", image: "humidity")
+                    Spacer()
+                    Label("\(String(format: "%.3f", model.pressure)) mBar", image: "pressure")
```
Should we got the value already with the unit text here?

kryzhikh replied:
updated along with using Measurement instead of custom units entities

MariaKulkova commented on WeatherUI/Views/CityWeatherView/CityWeatherView.swift:
```diff
@@ -0,0 +1,98 @@
+            VStack(alignment: .leading, spacing: 0.0) {
+                HStack {
+                    Text(model.cityName)
+                        .font(.system(size: 18, weight: .regular))
```
`regular` is the default value and could be omitted

kryzhikh replied:
updated, also added custom font as per design

MariaKulkova reviewed (approved):
Looks good! Left some minor comments

kotofey committed:
CityWeatherView and project updates

- Custom font added
- Measurement was used instead of custom entities
- Minor fixes
